docs/source/blocks/blocks-core/teaser.md

```diff
@@ -0,0 +1,152 @@
+
+The data adapter function is defined in the block's settings `dataAdapter`, so you can override it and add your own, if required.
+This is the default adapter (you should stick to his signature):
+
+```js
+import { isEmpty } from 'lodash';
+
+export const TeaserBlockDataAdapter = ({
+  block,
+  data,
+  id,
+  onChangeBlock,
+  value,
+}) => {
+  let dataSaved = {
+    ...data,
+    [id]: value,
+  };
+  if (id === 'href' && !isEmpty(value) && !data.title && !data.description) {
+    dataSaved = {
+      ...dataSaved,
+      title: value[0].Title,
+      description: value[0].Description,
+      head_title: value[0].head_title,
+    };
+  }
+  onChangeBlock(block, dataSaved);
+};
+
+```
+
+## Custom components in registry
+
+The Teaser block looks up for a couple of components in the registry, that you can provide in case that you want to modify the default behavior.
+
+### Image component
+
+The image component used in the Teaser can be overriden using your own custom image component.
+By default, it uses a simple `<img />` tag, with this signature:
+
+```jsx
+<Image
+  src={hasImageComponent ? href : defaultImageSrc}
+  alt=""
+  loading="lazy"
+/>
+```
+
+### Content type dependant Body component
+
+The Teaser component supports specifying a custom component depending on the target content type.
+
+You can provide your own component registering a component like in your add-on config:
+
+```js
+import NewsItem from './components/Blocks/Teaser/NewsItem';
+
+...
+
+  config.registerComponent({
+    name: 'Teaser',
+    component: NewsItem,
+    dependencies: 'News Item',
+  });
+```
+
+The `name` must be `Teaser` but then, as a dependency, you must provide the name of the content type.
+
+After this, if you select a content type `News Item` as target of the teaser, the `Body` component used will be `NewsItem` instead of the default one.
+
+## Custom Body component
+
+The Teaser body provides a default `Body` component that can be overriden.
+Although it can be done via component shadowing, it is recommended that you override it using the `variations` block settings field.
+
+```js
+import CustomTeaserBlockDefaultBody from './components/Blocks/Teaser';
+
+...
+
+    variations: [
+      {
+        id: 'default',
+        isDefault: true,
+        title: 'Default',
+        template: CustomTeaserBlockDefaultBody,
+      },
+    ],
+```
+
+The custom content type `Body` component mentioned in the previous section is compatible with this customization.
+In this case, the content type customization will take precedence, if it matches with the content type of the target.
+
+## Migration from `volto-blocks-grid`
+
+From Volto 16.14.0 the Teaser block as it's in `volto-blocks-grid` add-on version 7.0.0 is included in Volto core.
+
+Because of the add-on config priority, if you have `volto-blocks-grid` installed in your project, the Teaser block from the add-on (and your project amendments to the config, if any) will be applied and used.
```

Comment:
```suggestion
Because of the add-on configuration priority, if you have `volto-blocks-grid` installed in your project, the Teaser block from the add-on (and your project amendments to the configuration, if any) will be applied and used.
```